PubSub/src/Schema.php

```diff
@@ -241,33 +286,69 @@ public function listRevisions(array $options = [])
      */
     public function commit($definition, $type, array $options = [])
     {
-        return $this->connection->commitSchema([
-                'schema' => [
-                    'definition' => $definition,
-                    'type' => $type,
-                ],
-                'name' => $this->name
-            ] + $options);
+        $schemaObj = new SchemaProto([
+            'name' => $this->name,
+            'definition' => $definition,
+            'type' => Type::value($type),
+        ]);
+
+        $data = ['name' => $this->name, 'schema' => $schemaObj];
+        $request = $this->serializer->decodeMessage(new CommitSchemaRequest(), $data);
+
+        return $this->requestHandler->sendRequest(
+            SchemaServiceClient::class,
+            'commitSchema',
+            $request,
+            $options
+        );
     }
 
     /**
-     * Commit schema revision.
+     * Delete a schema revision
      *
      * Example:
      * ```
-     * $definition = file_get_contents('my-schema.txt');
-     * $revision = $schema->commit($definition, 'AVRO);
+     * $schema->delete($revisionId);
      *
      * @see https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.schemas/deleteRevision Delete revision.
      * ```
      *
      * @param string $revisionId The revisionId
      * @return array deleted revision
      */
-    public function deleteRevision($revisionId)
+    public function deleteRevision($revisionId, $options = [])
     {
-        return $this->connection->deleteRevision([
-            'name' => $this->name .'@' . $revisionId
-        ]);
+        $revisionName = $this->name .'@' . $revisionId;
+
+        $request = $this->serializer->decodeMessage(
+            new DeleteSchemaRevisionRequest(),
+            ['name' => $revisionName]
+        );
+
+        return $this->requestHandler->sendRequest(
+            SchemaServiceClient::class,
+            'deleteSchemaRevision',
+            $request,
+            $options
+        );
+    }
+
+    /**
+     * Helper function to return the value of 'view' from the options array.
+     * If a view key isn't specified, the default is returned.
+     *
+     * @param array $options The options array to extract the view from.
+     *
+     * @return int The integer value of the view specified as per Google\Cloud\PubSub\V1\SchemaView
+     */
+    private function getViewFromOptions(array &$options)
+    {
+        $view = $this->pluck('view', $options, false) ?? self::DEFAULT_VIEW;
```

Comment:
nit: the null-coalescing operator is specifically "unset" values (e.g. `$foo['bar'] ?? 'default-bar'` where we want it to be OKAY that the `bar` key isn't set, or even for the `$foo` variable to not exist). In this case, we _know_ the return value from pluck will exist, so it's best to just use the short inline ternary operator:
```suggestion
        $view = $this->pluck('view', $options, false) ?: self::DEFAULT_VIEW;
```